CHANGELOG.md

```diff
@@ -1,3 +1,60 @@
+# 0.4.0 TBA
+Features:
+* Added new `Dot` layer and `Conv1DTranspose`, `Conv2DTranspose`, `Conv3DTranspose` layers. 
+  [#144](https://github.com/Kotlin/kotlindl/issues/144) [#124](https://github.com/Kotlin/kotlindl/issues/124)
+* Added new activation functions: `SparsemaxActivation` and `SoftShrinkActivation`. 
+  [#171](https://github.com/Kotlin/kotlindl/issues/171) [#170](https://github.com/Kotlin/kotlindl/issues/170)
+* Added new `Padding`, `CenterCrop`, `Convert`, `Grayscale` image preprocessors and `Normalizing` tensor preprocessor.
+  [#203](https://github.com/Kotlin/kotlindl/issues/203) [#201](https://github.com/Kotlin/kotlindl/issues/201)
+  [#202](https://github.com/Kotlin/kotlindl/issues/202) [#204](https://github.com/Kotlin/kotlindl/issues/204)
+* Introduced new DSL for creating `Sequential` and `Functional` models. [#133](https://github.com/Kotlin/kotlindl/issues/133)
+
+New models:
+* Added NoTop models to the TensorFlow Model Hub. [#281](https://github.com/Kotlin/kotlindl/issues/281)
+* Added EfficientNet model family to the ONNX Model Hub. [#264](https://github.com/Kotlin/kotlindl/issues/264)
+* Added PoseNet and EfficientDet model families to the ONNX Model Hub. 
+  [#269](https://github.com/Kotlin/kotlindl/issues/269) [#304](https://github.com/Kotlin/kotlindl/issues/304)
+* Added SSD-Mobilenet-v1 model to the ONNX Model Hub [#296](https://github.com/Kotlin/kotlindl/issues/296)
+
+Examples and tutorials:
+* Added a tutorial for ResNet'50 fine-tuning.
+* Added an example of Image preprocessing DSL usage with TensorFlow model. [#292](https://github.com/Kotlin/kotlindl/issues/292)
+
+API changes:
+* Introduced new abstraction for layer parameters -- `KVariable`. [#324](https://github.com/Kotlin/kotlindl/pull/324)
```

Comment:
It's internal API changes: we need to separate internal API changes from external and probably add information about breaking changes based on comparision of 0.3 and 0.4 versions (how to migrate)